### PR TITLE
[v5.0.1] Remove v4.x implementation from Interpreter

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR the Interpreter is cleaned up from v4.x AOT costing and old reduction operations. Thus, only v5.0 implementation remains. 

The tests are adjusted to new cost values, but should remain exactly the same.

Note, the old AOT code will be cleaned up in separate PR.

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala [hunk 1]
```diff
@@ -625,12 +480,11 @@ object Interpreter {
 
   /** Result of fullReduction to sigma tree with costing. */
   case class FullReductionResult(
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:482]: This class looks like excessive meaningless wrapping now

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala [hunk 2]
```diff
@@ -1,7 +1,6 @@
 package sigmastate.interpreter
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:2]: A lot of unused imports now

**Review by kushti — APPROVED** [review]
Why expected costs changed in tests? Please explain

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:482]: Will be removed in subsequent PRs

**Comment by aslesarenko** [comment]
> Why expected costs changed in tests? Please explain

Explained this in ScalaDoc here https://github.com/ScorexFoundation/sigmastate-interpreter/blob/74d49fb171ff8236510f283d5e22c703c716985f/sigmastate/src/test/scala/special/sigma/SigmaDslSpecification.scala#L49

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala (hunk 2 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:2]: Cleaned up.

**Pull request merged**